argoverse/utils/make_att_files.py

```diff
@@ -0,0 +1,365 @@
+                    dict_tracks[id_track]["list_center_w"]
+                )
+                dict_tracks[id_track]["num_missing"] = (
+                    dict_tracks[id_track]["length_track"] - dict_tracks[id_track]["exists"].sum()
+                )
+                dict_tracks[id_track]["difficult_att"] = []
+                vel_abs = np.linalg.norm(dict_tracks[id_track]["list_vel"][:, 0:2], axis=1)
+                acc_abs = np.linalg.norm(dict_tracks[id_track]["list_acc"][:, 0:2], axis=1)
+
+                dist_close = 50
+                ind_valid = np.nonzero(1 - np.isnan(dict_tracks[id_track]["list_dist"]))[0]
+                ind_close = np.nonzero(dict_tracks[id_track]["list_dist"][ind_valid] < dist_close)[0]
+
+                if len(ind_close) > 0:
+                    ind_close_max = ind_close.max() + 1
+                    ind_close_min = ind_close.min()
+
+                if dict_tracks[id_track]["length_track"] == 0:
+                    # this shouldn't happen
+                    dict_tracks[id_track]["difficult_att"].append("zero_length")
+
+                else:
+                    if dict_tracks[id_track]["list_dist"][ind_valid].min() > dist_close:
+                        dict_tracks[id_track]["difficult_att"].append("far")
+                    else:
+                        if dict_tracks[id_track]["length_track"] < 10 or dict_tracks[id_track]["exists"].sum() < 10:
+                            dict_tracks[id_track]["difficult_att"].append("short")
+                        else:
+                            if (ind_close_max - ind_close_min) - dict_tracks[id_track]["exists"][
+                                ind_close_min:ind_close_max
+                            ].sum() > 20:
+                                dict_tracks[id_track]["difficult_att"].append("occ")
+
+                            if np.quantile(vel_abs[ind_valid][ind_close], 0.9) > 1:
+                                dict_tracks[id_track]["difficult_att"].append("fast")
+
+                    if len(dict_tracks[id_track]["difficult_att"]) == 0:
+                        dict_tracks[id_track]["difficult_att"].append("easy")
+
+            if visualize:
+                for ind_lidar, timestamp_lidar in enumerate(list_lidar_timestamp):
+
+                    list_bboxes = []
+                    list_difficulty_att = []
+
+                    for id_track in dict_tracks.keys():
+                        if dict_tracks[id_track]["exists"][ind_lidar]:
+                            list_bboxes.append(dict_tracks[id_track]["list_bbox"][ind_lidar])
+                            list_difficulty_att.append(dict_tracks[id_track]["difficult_att"])
+
+                    path_lidar = os.path.join(path_log, "lidar", "PC_%s.ply" % timestamp_lidar)
+                    egovehicle_to_city_se3 = argoverse_loader.get_pose(ind_lidar, id_log)
+                    pc = np.asarray(o3d.io.read_point_cloud(path_lidar).points)
```

Comment:
does open3d support reading the reflectance values now? We could deprecate pyntcloud and switch to open3d altogether if so

Reply:
Maybe we could use the xyzirgb format here? http://www.open3d.org/docs/release/tutorial/Basic/file_io.html#Point-cloud